### DISCUSSION TITLE
API for relationship subfields how to get the relationship records number in $this->crud->entry ...?

### DISCUSSION THREAD

**Discussion by lieto-dk** [Laravel-Backpack/community-forum#410] [Q&A (Help)]
Hi, I have a field in my cruds setupcreateoperation with a relationship and subfields where i want one of the fields to be a select2 with options that alters the query with a fields value in the subfields relationship record.

The field with subfields is defined as , 
```php
$this->crud->addField([
            'name' => "faqs",
            'label' => 'Frequently Asked Questions (FAQ) per site',
            'tab' => 'FAQ',
            'type' => 'relationship',
            'new_item_label'  => 'Add FAQ',
            'init_rows' => 0,
            'subfields'   => [
```

And the subfield should be something like 
```php
  [
                    'name'  => 'faqautoquestion_id',
                    'type'  => 'select2',
                    'label' => 'Faq Auto Question',
                    'model' => 'App\Models\Faqautoquestion',
                    'attribute' => 'question',
                    'options'   => (function ($query) {
                        return $query->where('faqable_type', 'App\Models \Boardtype')->where('site_id',$this->crud->entry->faqs[X]->site_id)->get();
                    }),
                    'wrapper'   => ['class' => 'form-group col-md-12'],
                ],
```

My problem is how do i get the X value (the number of the record that i'm editing in the "relationship") ?

**Comment by lieto-dk** [comment]
Hmm .. no one with any suggestions how to solve this ?

**Reply by o15a3d4l11s2** [reply]
The question is a little unclear to me. Do you need to get the count of `faqautoquestion_id`s during save/update CRUD operation, while opening the edit CRUD, or you need them later on in another place?

From your last sentence it seems that you might need them live while the edit is happening?

**Comment by lieto-dk** [comment]
Hi, i need the records number live as i use it to filter the selection for a field in the record based on another fields value in the record.
Any suggestions on how to solve this ? Is there a API call that gets the current record ?

**Reply by o15a3d4l11s2** [reply]
Based on your reply I believe the best option would be to use a JS event while the editing is happening.
For more details please refer to https://backpackforlaravel.com/docs/5.x/crud-fields-javascript-api#examples-1.
The whole feature is described also in the blog: https://backpackforlaravel.com/articles/news/new-feature-crudfield-javascript-library

**Reply by tabacitu** [reply]
Oooh yeah, sounds like CrudField JS Library is the way to go. That will give you the ability to add JS on the form... target the `relationship` field and any of its subfields... interpret its current value... and do whatever you want with it (in JS).

Good catch @o15a3d4l11s2 👍

**Comment by lieto-dk** [comment]
Thx for the answers. Will give it a try - but would have loved if the method for setting options was the same in subfields as in regular fields. Javascript is fine but not really needed for my requirements if the $this->crud->entry ... was callable for the current record in the subfields. Just my penny!

**Reply by o15a3d4l11s2** [reply]
It is not possible to use `$this->crud->entry` for dynamically changing the options of the subfield while you are editing it, as the `entry` is still not saved, so the data is not there. While the form is being edited the only way to get the selected/entered data is through JS (or submit it to the server through AJAX, but that is a completely different topic).

Maybe I am still not understanding you use case completely. Feel free to take a short demo video of the usage and point exactly what data is missing and where. Maybe this way somebody will be able to help with more specific information.

**Comment by jcastroa87** [comment]
Hello,

We haven’t heard back from you in a while, so we’re closing this issue for now. If you still need assistance, please don’t hesitate to reopen the issue or create a new one.

Thank you, and take care!

Cheers.
